tests/conftest.py

```diff
@@ -461,7 +461,19 @@ def kill_all_existing_windows(show_warnings=True):
         subprocess.run(['xdotool', 'windowkill', c])
 
 
-@pytest.fixture(autouse=True)
+@pytest.fixture()
+def hlwm_spawner(tmpdir):
```

Comment:
Why not make use of `hlwm_spawner` in `hlwm_process`? Putting together the environment and creating the HlwmProcess is basically the same code in both fixtures now, isn't it?

Reply:
Bascially yes, I suspected that it makes `hlwm_process` more nested&complicated to read but not really shorter. I only wrote `hlwm_spawner` because I didn't know how to access the constructor `HlwmProcess()` from test cases directly.

I've made this change in another branch: https://github.com/herbstluftwm/herbstluftwm/commit/ed171c854076b32c1b5430cc250bef3033a9bccb Do you think this change is an improvement. I'm really not sure.

Reply:
I think it's pretty logical. It's making use of the building blocks that we provide.

Would there be any reason not to move the assert into hlwm_spawner as well?

Reply:
The assert is only there to avoid that clients of an actual session are killed. But we also can assert this for hlwm_spawner, it does not really matter, I think.

So I made all these changes in the present PR